eclipse-theia/theia
# Implement CancellationToken logic for withProgress API

#### What it does

- Unlocks functionality of `token.onCancellationRequested` & `token.isCancellationRequested` (from the VS Code Extension API)
- Fixes https://github.com/eclipse-theia/theia/issues/11030 
#### How to test

- Clone this [VS Code Example](https://github.com/microsoft/vscode-extension-samples/tree/main/progress-sample) and open in Theia
- Run the Launch Extension launch configuration
- Add a `console.log` within the `setTimeout` callbacks [here](https://github.com/microsoft/vscode-extension-samples/blob/2556c82cb333cf65d372bd01ac30c35ea1898a0e/progress-sample/src/extension.ts#L21-L31) to check the value of `token.isCancellationRequested`
- Run the extension's `showProgress` command (using the command pallette) and press the cancel button contained in the notification
- Pre-changes
    - No console output from within `token.onCancellationRequested`
    - `token.isCancellationRequested` remains false indefinitely
 - Post-changes
    - Console output from within `token.onCancellationRequested`
    - `token.isCancellationRequested` changes based on button press

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Matthew Langley <matthew.langley@arm.com>
